external-plugins/rehearse/events/events.go

```diff
@@ -0,0 +1,397 @@
+
+func validateDryRunEvent(w http.ResponseWriter, r *http.Request, t func() []byte) (string, string, []byte, bool, int) {
+	if r.Method != http.MethodPost {
+		return "", "", nil, false, 0
+	}
+	eventType := r.Header.Get("X-GitHub-Event")
+	if eventType == "" {
+		return "", "", nil, false, 0
+	}
+	eventGUID := r.Header.Get("X-GitHub-Delivery")
+	if eventGUID == "" {
+		return "", "", nil, false, 0
+	}
+	sig := r.Header.Get("X-Hub-Signature")
+	if sig == "" {
+		return "", "", nil, false, 0
+	}
+	contentType := r.Header.Get("content-type")
+	if contentType != "application/json" {
+		return "", "", nil, false, 0
+	}
+	payload, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return "", "", nil, false, 0
+	}
+
+	return eventType, eventGUID, payload, true, 0
+}
+
+func (s *GitHubEventsServer) onUpdateEvent(logger *logrus.Entry, event *github.PullRequestEvent) {
+	logger.Infof("Handling updated pull request: %s [%d]", event.Repo.FullName, event.PullRequest.Number)
+	repo, org, err := git2.OrgRepo(event.Repo.FullName)
+	if err != nil {
+		s.logger.WithError(err).Errorf("Could not parse repo name: %s", event.Repo.FullName)
+		return
+	}
+	git, err := s.gitClientFactory.ClientFor(repo, org)
+	if err != nil {
+		return
+	}
+	err = git.FetchRef(event.PullRequest.Base.Ref)
+	if err != nil {
+		s.logger.WithError(err).Error("Could not fetch pull request's base ref.")
+		return
+	}
+	err = git.FetchRef(event.PullRequest.Head.Ref)
+	if err != nil {
+		s.logger.WithError(err).Error("Could not fetch pull request's head ref.")
+		return
+	}
+	changedFiles, err := git.Diff(event.PullRequest.Head.Ref, event.PullRequest.Base.Ref)
+	if err != nil {
+		s.logger.WithError(err).Error("Could not calculate diff for PR.")
+		return
+	}
+	changedJobConfigs, err := s.absModifiedJobConfigs(changedFiles)
+	if err != nil {
+		s.logger.WithError(err).Error("Could not calculate absolute paths for modified job configs")
+		return
+	}
+	headConfigs, err := s.loadConfigsAtRef(changedJobConfigs, git, event.PullRequest.Head.Ref)
+	if err != nil {
+		s.logger.WithError(err).Errorf(
+			"Could not load job configs from head ref: %s", event.PullRequest.Head.Ref)
+	}
+	baseConfigs, err := s.loadConfigsAtRef(changedJobConfigs, git, event.PullRequest.Base.Ref)
+	if err != nil {
+		s.logger.WithError(err).Errorf(
+			"Could not load job configs from base ref: %s", event.PullRequest.Base.Ref)
+	}
+
+	prowjobs := s.generateProwJobs(headConfigs, baseConfigs, event)
+	if s.dryRun {
+		for _, job := range prowjobs {
+			s.logger.Infof("%+v", job)
+		}
+		return
+	}
+}
+
+func (s *GitHubEventsServer) generateProwJobs(
+	headConfigs, baseConfigs map[string]*config.Config, event *github.PullRequestEvent) []prowapi.ProwJob {
+	var jobs []prowapi.ProwJob
+
+	for path, headConfig := range headConfigs {
+		baseConfig, _ := baseConfigs[path]
+		jobs = append(jobs, s.generatePresubmits(headConfig, baseConfig, event)...)
+		jobs = append(jobs, s.generatePeriodics(headConfig, baseConfig, event)...)
+	}
```

Comment:
Post submits? Not yet I guess?

Reply:
Not yet. I actually also removed the support for periodics for now, but we'll add it back soon.